pkg/volume/csi/csi_attacher.go

```diff
@@ -166,6 +171,43 @@ func (c *csiAttacher) waitForVolumeAttachmentInternal(volumeHandle, attachID str
 	return attach.Name, nil
 }
 
+func (c *csiAttacher) waitForVolumeAttachmentWithLister(volumeHandle, attachID string, pollInterval, timeout time.Duration) error {
```

Comment:
What is the timeout? If it's too long, then we may end up excessively spinning too much if the apiserver is slow/having problems, especially with multiple attachements happening in parallel. This is the main reason why we do exponential backoff.

In general we've tried to move away from having retry loops within retry loops because it makes it difficult to reason about the behavior.

Reply:
In your benchmarking test, it would be good to also inject a sleep into the mock driver to simulate a more realistic volume plugin with a longer attach time.

Reply:
If needed, we could probably double the pollInterval after every tick. If we do that we can probably start with small interval and exponentially increase after every tick.

Reply:
Discussed this offline. The reasons for keeping this synchronous for now was in order to stay consistent with the attachment behavior used by our other (in-tree) plugins, and also because making this asynchronous adds a lot of complexity to our tests. That work is probably best tackled in a separate PR.

Please see the comment below on thoughts around using exponential backoff vs. fixed poll intervals.

Reply:
Also for context, I tried adding a `time.Sleep(40 * time.Second)` in the mock CSI driver `ControllerPublishVolume()` call, however I encountered other issues with the mock CSI driver that prevented this experiment from working. This requires more digging to understand what's going on.